### PR TITLE
Add `scarb new` and `scarb init` commands, getting started in README

### LINKED ISSUE #30: Implement `new` and `init` commands

### LINKED ISSUE #32: Document installation and getting started in readme

### PR DESCRIPTION
**Pull request by mkaput**
fix #30
fix #32

### CODE REVIEW

File: scarb/src/ops/new.rs [hunk 1]
```diff
@@ -0,0 +1,152 @@
...
+            indoc! {r#"
+                fn fib(a: felt, b: felt, n: felt) -> felt {
+                    match n {
+                        0 => a,
+                        _ => fib(b, a + b, n - 1),
+                    }
+                }
+            "#},
+        )?;
+    }
```
Comment (maciektr) [inline scarb/src/ops/new.rs:139]: On a similar note, should we overwrite source files that may be created by user before using `init`? I presume one may want to use `init` on a project that has already been populated with some source files.

File: scarb/tests/new_and_init.rs [hunk 2]
```diff
@@ -0,0 +1,136 @@
...
+        .arg("world")
+        .current_dir(&pt)
+        .assert()
+        .success();
+
+    let t = pt.child("hello");
+
+    let toml_manifest = TomlManifest::read_from_path(t.child("Scarb.toml").path()).unwrap();
+    assert_eq!(toml_manifest.package.unwrap().name, "world");
+}
```
Comment (maciektr) [inline scarb/tests/new_and_init.rs:136]:
```suggestion
}

#[test]
fn init_existing_manifest() {
    let pt = assert_fs::TempDir::new().unwrap();
    let t = pt.child("hello");
    t.create_dir_all().unwrap();

    let manifest_path = t.path().join("Scarb.toml");
    let mut manifest_file = File::create(manifest_path.clone()).unwrap();
    writeln!(manifest_file, "Scarb is great!").unwrap();

    Command::new(cargo_bin!("scarb"))
        .arg("init")
        .current_dir(&t)
        .assert()
        .failure();

    assert!(t.is_dir());
    assert!(!t.child("src/lib.cairo").is_file());
    assert!(!t.child(".gitignore").is_file());

    let manifest_contents =
        fs::read_to_string(manifest_path).expect("Should have been able to read the file");
    assert_eq!(manifest_contents, "Scarb is great!");

    drop(manifest_file);
}
```

Maybe? :open_hands:

**Review by maciektr — APPROVED** [review]

File: scarb/src/ops/new.rs (hunk 1 above)
Comment (mkaput) [reply scarb/src/ops/new.rs:139]: There's `    if !source_dir.exists() {` to check this

File: scarb/src/ops/new.rs [hunk 3]
```diff
@@ -0,0 +1,152 @@
...
+    path: PathBuf,
+    name: PackageName,
+}
+
+fn mk(MkOpts { path, name }: MkOpts, _config: &Config) -> Result<()> {
+    // Create project directory in case we are called from `new` op.
+    fsx::create_dir_all(&path)?;
+
+    // Write VCS ignore file.
+    fsx::write(path.join(".gitignore"), DEFAULT_TARGET_DIR_NAME)?;
```
[the remark this answers is not in the document]
Comment (mkaput) [reply scarb/src/ops/new.rs]: https://github.com/software-mansion/scarb/pull/36/commits/74036b7ff49334fb789a7ee967419b202e855803

File: scarb/src/ops/new.rs (hunk 1 above)
Comment (mkaput) [reply scarb/src/ops/new.rs:143]: https://github.com/software-mansion/scarb/pull/36/commits/2a56e8479d933c8a657ec7a9fb6cb500db9ee31e

**Review by maciektr — APPROVED** [review]

**Pull request merged**